### PR TITLE
[fix]: [SMP-1003]: remove singleServerConfig if sentinels is enabled 

### PR DESCRIPTION
**Pull request by arya-harness**
## Describe your changes
The PR is for a regression in 776xx caused by using REDIS_NETTY_THREAD in configmap. Inclusion of this variable was not tested with redisSentinels and lead to crash.
It contains wrapping of singleServerConfig which were previously unrestricted and added to sentinels lead to pod crashes.
Additionally fixed redisDelegateConfig not working if sentinels are added, and fixed a typo in manager's replace_config.yaml in redisDelegateConfig for sentinels.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44925)

### CODE REVIEW

**Comment by arya-harness** [comment]
trigger feature-build

**Comment by arya-harness** [comment]
trigger messagecheck

File: 420-delegate-service/src/main/java/io/harness/service/impl/DelegateRingServiceImpl.java [hunk 1]
```diff
@@ -1,3 +1,10 @@
+/*
```
Comment (arya-harness) [inline 420-delegate-service/src/main/java/io/harness/service/impl/DelegateRingServiceImpl.java:1]: Got created on pre-commit hooks. (unrelated to PR)

**Comment by arya-harness** [comment]
trigger messagecheck

**Comment by arya-harness** [comment]
trigger feature-build

**Comment by arya-harness** [comment]
trigger feature-build

**Review by akshay-khole — APPROVED** [review]

**Review by jennyjames21 — APPROVED** [review]
> Reviewed 1 of 3 files at r1, 1 of 2 files at r2, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44925)* status: Waiting on team(s): del-team.

**Comment by kapilgarg1996** [comment]
@ashishsanodia-harness can you review it once ? We need to push this hotfix by monday EOD

**Review by kapilgarg1996 — APPROVED** [review]

**Pull request merged**